lib/Basics/Thread.cpp

```diff
@@ -89,33 +88,21 @@ void Thread::startThread(void* arg) {
     TRI_SetProcessorAffinity(&ptr->_thread, ptr->_affinity);
   }
 
-  bool pushed = WorkMonitor::pushThread(ptr);
-
   try {
```

Comment:
can be simplified by using `TRI_DEFER(ptr->cleanupMe());` here.

Reply:
why add another lambda to every thread ?